test/integration/etcd/server.go

```diff
@@ -125,23 +125,21 @@ func StartRealMasterOrDie(t *testing.T, configFuncs ...func(*options.ServerRunOp
 
 	kubeClient := clientset.NewForConfigOrDie(kubeClientConfig)
 
-	go func() {
-		// Catch panics that occur in this go routine so we get a comprehensible failure
-		defer func() {
-			if err := recover(); err != nil {
-				t.Errorf("Unexpected panic trying to start API master: %#v", err)
-			}
-		}()
-
-		prepared, err := kubeAPIServer.PrepareRun()
-		if err != nil {
-			t.Fatal(err)
-		}
-		if err := prepared.Run(stopCh); err != nil {
-			t.Fatal(err)
+	// Catch panics that occur in this go routine so we get a comprehensible failure
+	defer func() {
+		if err := recover(); err != nil {
+			t.Errorf("Unexpected panic trying to start API master: %#v", err)
 		}
 	}()
 
+	prepared, err := kubeAPIServer.PrepareRun()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := prepared.Run(stopCh); err != nil {
```

Comment:
this is not equivalent. Run() blocks, so moving this outside the goroutine will prevent the test from running/completing

Reply:
So i need to keep goroutine When fixing this error?

Reply:
I made a modified submission please review it. thinks!